pilot/pkg/serviceregistry/kube/controller/ambient/remotesecrets.go

```diff
@@ -0,0 +1,151 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ambient
+
+import (
+	"crypto/sha256"
+	"sync/atomic"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/rest"
+
+	"istio.io/api/label"
+	"istio.io/istio/pilot/pkg/features"
+	"istio.io/istio/pkg/cluster"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/kube/kclient"
+	"istio.io/istio/pkg/kube/krt"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/monitoring"
+	"istio.io/istio/pkg/network"
+	"istio.io/istio/pkg/ptr"
+)
+
+const (
+	MultiClusterSecretLabel = "istio/multiCluster"
+)
+
+var (
+	clusterLabel = monitoring.CreateLabel("cluster")
+	timeouts     = monitoring.NewSum(
+		"ambient_remote_cluster_sync_timeouts_total",
+		"Number of times remote clusters took too long to sync, causing slow startup that excludes remote clusters.",
+	)
+
+	clusterType = monitoring.CreateLabel("ambient_cluster_type")
+
+	clustersCount = monitoring.NewGauge(
+		"ambient_istiod_managed_clusters",
+		"Number of clusters managed by istiod",
+	)
+
+	localClusters  = clustersCount.With(clusterType.Value("local"))
+	remoteClusters = clustersCount.With(clusterType.Value("remote"))
+)
+
+// ClientBuilder builds a new kube.Client from a kubeconfig. Mocked out for testing
+type ClientBuilder = func(kubeConfig []byte, clusterId cluster.ID, configOverrides ...func(*rest.Config)) (kube.Client, error)
+
+func buildRemoteClustersCollection(
+	options Options,
+	opts krt.OptionsBuilder,
+	builder ClientBuilder,
+	filter kclient.Filter,
+	configOverrides ...func(*rest.Config),
+) krt.Collection[Cluster] {
+	informerClient := options.Client
+
+	// When these two are set to true, Istiod will be watching the namespace in which
+	// Istiod is running on the external cluster. Use the inCluster credentials to
+	// create a kubeclientset
+	if features.LocalClusterSecretWatcher && features.ExternalIstiod {
+		config, err := kube.InClusterConfig(configOverrides...)
+		if err != nil {
+			log.Errorf("Could not get istiod incluster configuration: %v", err)
+			return nil
+		}
+		log.Info("Successfully retrieved incluster config.")
+
+		localKubeClient, err := kube.NewClient(kube.NewClientConfigForRestConfig(config), options.ClusterID)
+		if err != nil {
+			log.Errorf("Could not create a client to access local cluster API server: %v", err)
+			return nil
+		}
+		log.Infof("Successfully created in cluster kubeclient at %s", localKubeClient.RESTConfig().Host)
+		informerClient = localKubeClient
+	}
+
+	// init gauges
+	localClusters.Record(1.0)
+	remoteClusters.Record(0.0)
+
+	secrets := kclient.NewFiltered[*corev1.Secret](informerClient, kclient.Filter{
+		Namespace:     options.SystemNamespace,
+		LabelSelector: MultiClusterSecretLabel + "=true",
+	})
+	Secrets := krt.WrapClient(secrets, opts.WithName("RemoteSecrets")...)
+
+	Clusters := krt.NewManyFromNothing(func(ctx krt.HandlerContext) []Cluster {
```

Comment:
Should this be `New`? Regardless of the collection type, I'm a little worried about events that don't actually change the info we care about making us rebuild state for clusters. 

https://github.com/istio/istio/blob/main/pkg/kube/multicluster/secretcontroller.go#L286

At the same time, I get that this is WIP, I was just looking to better understand the other KRT ones.

Reply:
Ah this is leftover from a previous iteration where I was appending the local cluster to the clusters collection. Good catch!